src/Core/UUID.h

```diff
@@ -2,6 +2,59 @@
 
 #include <Core/Types.h>
 
+/**
+ * Implementation Details
```

Comment:
@rschu1ze I liked your idea of repurposing the documentation I wrote in the other PR here. I have adjusted what I had to be correct for this context.

Please let me know if there are any formatting changes that you'd like to see made 😁